src/userspace/monitor.hpp

```diff
@@ -0,0 +1,11 @@
+#ifndef MONITOR_H
+#define MONITOR_H
+
+class Monitor {
+private:
+	virtual int export_data() = 0;
```

Comment:
Since it's an interface and there is no implementation, it's hard to tell what this does. Could you add a comment to both functions so that someone who implements a Monitor knows what to do?